app/src/main/java/com/matheus/starwars/adapter/PeopleAdapter.kt

```diff
@@ -0,0 +1,52 @@
+package com.matheus.starwars.adapter
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import com.matheus.starwars.R
+import com.matheus.starwars.databinding.ItemPeopleBinding
+import com.matheus.starwars.model.People
+
+class PeopleAdapter(private var PeopleList: ArrayList<People> ) :
+    RecyclerView.Adapter<PeopleAdapter.PeopleViewHolder>() {
+
+    class PeopleCallback(
+        peopleList: ArrayList<People>,
+        listOfPeoples: List<People>
+    ) : DiffUtil.ItemCallback<People>() {
+        override fun areItemsTheSame(oldItem: People, newItem: People): Boolean {
+            return oldItem == newItem
+        }
+
+        override fun areContentsTheSame(oldItem: People, newItem: People): Boolean {
+            return oldItem == newItem && oldItem == newItem
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PeopleViewHolder {
+        val itemPeopleBinding: ItemPeopleBinding = DataBindingUtil.inflate(
+            LayoutInflater.from(parent.context),
+            R.layout.item_people,
+            parent,
+            false
+        )
+        return PeopleViewHolder(itemPeopleBinding)
+    }
+
+    override fun getItemCount(): Int = PeopleList.size
+
+    override fun onBindViewHolder(holder: PeopleViewHolder, position: Int) {
+        holder.itemPeopleBinding.people = PeopleList[position]
+    }
+
+    fun setUpPeoples(listOfPeoples: List<People>) {
+        PeopleList.clear()
+        PeopleList.addAll(listOfPeoples)
+
```

Comment:
Com o callback do `DiffUtil` implementado corretamente, bastava você implementar o seguinte trecho de código:

```suggestion
        val diffResult = DiffUtil.calculateDiff(PeopleDiffCallback(PeopleList, listOfPeoples))
        PeopleList = arrayListOf(*listOfPeoples.toTypedArray())
        diffResult.dispatchUpdatesTo(this)
``` 
Com isso, você está calculando o diff da lista antiga e nova, e despachando o resultado para o `PeopleAdapter` atualizar somente o necessário através do `DiffResult`, por isso é muito importante para evitar problemas de performance na `RecyclerView`.

Reply:
Acabei de ver que teu layout da MainActivity está "quebrado" lá no XML, tenta dá corrigida levando em consideração o que eu sugeri na última review.